applications/CompressiblePotentialFlowApplication/python_scripts/initialize_geometry.py

```diff
@@ -0,0 +1,252 @@
+        KratosMultiphysics.Logger.PrintInfo('InitializeGeometry','Elapsed time: ',time.time()-ini_time)
+
+        ''' ############################################################################################## '''
+        ''' THIS FUNCTION CALL IS TEMPORARY AND WILL BE REMOVED ONCE THE EMBEDDED WAKE PROCESS IS DEFINED '''
+        # Find nodal neigbours util call
+        avg_elem_num = 10
+        avg_node_num = 10
+        KratosMultiphysics.FindNodalNeighboursProcess(
+            self.main_model_part, avg_elem_num, avg_node_num).Execute()
+        ''' ############################################################################################## '''
+
+    def ExecuteInitialize(self):
+        self.Execute()
+
+    def InitializeSkinModelPart(self):
+        ''' This function loads and moves the skin_model_part in the main_model_part to the desired initial point (self.initial_point).
+            It also rotates the skin model part around the initial point according to the (self.geometry_parameter)'''
+        self.skin_model_part=self.model.CreateModelPart("skin")
+        KratosMultiphysics.ModelPartIO(self.skin_model_part_name).ReadModelPart(self.skin_model_part)
+        ini_time=time.time()
+        if self.skin_model_part_name=='naca0012':
+            MoveModelPart(self.initial_point, self.skin_model_part)
+            self.origin=[0.25+self.initial_point[0],0+self.initial_point[1]]
+            angle=math.radians(-self.geometry_parameter)
+            RotateModelPart(self.origin,angle,self.skin_model_part)
+        elif self.skin_model_part_name=='ellipse':
+            MoveModelPart(self.initial_point, self.skin_model_part)
+            self.origin=[self.initial_point[0],self.initial_point[1]]
+            angle=math.radians(-self.geometry_parameter)
+            RotateModelPart(self.origin,angle,self.skin_model_part)
+        KratosMultiphysics.Logger.PrintInfo('InitializeGeometry','InitializeSkin time: ',time.time()-ini_time)
+
+    def CalculateDistance(self):
+        ''' This function calculate the distance to skin for every node in the main_model_part.'''
+        ini_time=time.time()
+        KratosMultiphysics.CalculateDistanceToSkinProcess2D(self.main_model_part, self.skin_model_part).Execute()
+        KratosMultiphysics.Logger.PrintInfo('InitializeGeometry','CalculateDistance time: ',time.time()-ini_time)
+
+    def ExtendDistance(self):
+        ''' This function extends the distance field to all the nodes of the main_model_part in order to
+            remesh the background mesh.'''
+        ini_time=time.time()
+        # Construct the variational distance calculation process
+        maximum_iterations = 2 #TODO: Make this user-definable
+        if self.main_model_part.ProcessInfo[KratosMultiphysics.DOMAIN_SIZE] == 2:
+            variational_distance_process = KratosMultiphysics.VariationalDistanceCalculationProcess2D(
+                self.main_model_part,
+                self.linear_solver,
```

Comment:
If I'm not wrong, you only need the linear solver in here. You can avoid to save it as a member variable if you construct it in here.